### PR TITLE
Property DDS 

### PR DESCRIPTION
**Pull request by arno-123**
This PR introduces Property DDS, a tree data structure for concurrent editing.
It enables git-like workflows and has components that facilitate data binding and debugging.
Property DDS supports dynamic collections as well as statically typed data. 

@SamBroner 
@DLehenbauer 
@tylerbutler 
PTAL

@ruiterr 
@karlbom 
@nedalhy 
@evaliyev 
FYI

### CODE REVIEW

**Comment by DLehenbauer** [comment]
FYI - I tracked down the cause of the build failures in the ODSP packages and have a fix pending (see #5788).

File: packages/drivers/tinylicious-driver/src/insecureTinyliciousUrlResolver.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ import { ITokenClaims } from "@fluidframework/protocol-definitions";
 import { KJUR as jsrsasign } from "jsrsasign";
 import { v4 as uuid } from "uuid";
 
-export const defaultTinyliciousPort = 35843;
+export const defaultTinyliciousPort = 7070;
```
Comment (DLehenbauer) [inline packages/drivers/tinylicious-driver/src/insecureTinyliciousUrlResolver.ts:16]: FYI - Hopefully you won't need this soon (see PR #5630).

Comment (arno-123) [reply packages/drivers/tinylicious-driver/src/insecureTinyliciousUrlResolver.ts:16]: Thank you, @DLehenbauer  !

**Comment by karlbom** [comment]
currently building client packages fails in [@fluidframework/webpack-fluid-loader](https://dev.azure.com/fluidframework/public/_build/results?buildId=21741&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=3e8a0811-333e-5612-55d7-750fabab9520&l=1576) building the test in `src/test/dependencies.spec.ts`

**Comment by DLehenbauer** [comment]
Build should now complete after picking up the latest from 'main':

> git checkout arno-123/property_DDS_PR
> git merge origin main

...I was going to push this back into the 'property_DDS_PR', but wasn't sure of your workflow (e.g., are you rebasing and --force pushing to update the branch?)

You can get a preview of what's next by running this from root:
> npm run policy-check

...we're getting close.  :-)

**Comment by karlbom** [comment]
> Build should now complete after picking up the latest from 'main':
> 
> > git checkout arno-123/property_DDS_PR
> > git merge origin main
> 
> ...I was going to push this back into the 'property_DDS_PR', but wasn't sure of your workflow (e.g., are you rebasing and --force pushing to update the branch?)
> 
> You can get a preview of what's next by running this from root:
> 
> > npm run policy-check
> 
> ...we're getting close. :-)

thank you! I think usually we would rebase to keep the git history clean but since we do parallel work based on the PR branch, rebasing causes a lot of churn so for now I will rather go with the merge path.

**Comment by karlbom** [comment]
I have noticed the policy check hangs on large file, I tracked it down to the  copyright header checks the tests in

https://github.com/microsoft/FluidFramework/blob/main/tools/build-tools/src/repoPolicyCheck/handlers/copyrightFileHeader.ts#L62

`   if (!/(\/\/|[\s\S]*\*)[\s\S]*Copyright \(c\) Microsoft Corporation. All rights reserved./i.test(content)
                || !/(\/\/|[\s\S]*\*)[\s\S]*Licensed under the MIT License./i.test(content)) {
                return 'JavaScript/TypeScript file missing copyright header';
            }`


I suspect that these regex tests  run rather slow on large files (I saw it hang on a 2k lines file), since they should be headers one could take first n lines or characters to test for the presence of the copyright header instead of the whole file.
This could also improve overall performance of the policy checker

**Comment by DLehenbauer** [comment]
@karlbom - Sorry, I didn't get a chance to look at the poly-check hang today.  Hopefully tomorrow.

**Comment by DLehenbauer** [comment]
@karlbom - The hang in policy-check is fixed now (nice find!).

Also, Fluid released today, so the version numbers in our repo advanced to ^0.39.  (I've updated the dependencies in /PropertySet/... accordingly.)

Next is probably to look at the couple of build errors in 'Build - client packages'.

**Comment by ruiterr** [comment]
In the propertyDDS tests, I get a lot of warnings:

```
ContainerRuntime with AgentScheduler built-in is deprecated, see BREAKING.md for more details and migration instructions
The ContainerRuntime.leader property and "leader"/"notleader" events are deprecated, see BREAKING.md for more details and migration instructions
```

I am not aware, that I directly call the referenced interfaces. Can I ignore the warnings and they will be fixed in future releases?

**Comment by ChumpChief** [comment]
> I am not aware, that I directly call the referenced interfaces. Can I ignore the warnings and they will be fixed in future releases?

Yes, these are unrelated. #5833 will change the default in most cases which will suppress these.

**Comment by ruiterr** [comment]
> @karlbom - The hang in policy-check is fixed now (nice find!).

@DLehenbauer  It seems that the build is again stuck in te policy-check. I changed the copyright messages in the headers, so maybe this could have triggered this?

**Comment by ruiterr** [comment]
I merged in the newes tip of main to get the updated policy-checker.

@nedalhy @evaliyev I had to make a few small changes to the propertyDDS branch in the merge commit. Since I created the merge commit directly, could you quickly check them? I mainly reverted a few changes because I now had to update back to async 3.2.

@DLehenbauer I only had to make one change that relates to other parts of fluid. I bumped the async version for @fluid-example/scribe to 3.2. I think we don't need to do any code changes for this example, but I was not able to try the example. It warned me that it is not possible to create docuements. Is this, because I am running against tinylicious?

**Comment by DLehenbauer** [comment]
@rueterr - You can ignore the 'scribe' example.  (It requires more infrastructure than just the webpack dev server.)

**Comment by DLehenbauer** [comment]
@ruiterr, @karlbom, @evaliyev - I *think* that 'policy-check' was stuck on a merge conflict with the Lerna package lock file.  I've pulled the latest from origin/head into 'arno-123/property_DDS_PR' and regenerated the file.

(The remaining policy-check issues can be fixed by running `npm run policy-check:fix` from root and committing the result.)

Client build was just missing some 'api-extractor.json' config files required by the 'build:docs' task.  (Fixed)

The Server build is hitting another instance of #5796.  I'll take a look at this next...

**Comment by ruiterr** [comment]
All checks have passed! 😃

**Review by DLehenbauer — APPROVED** [review]

**Pull request merged**